components/sidebar/sidebar.jsx

```diff
@@ -168,22 +170,35 @@ export default class Sidebar extends React.PureComponent {
         this.unreadScrollAnimate.addListener({onSpringUpdate: this.handleScrollAnimationUpdate});
     }
 
+    getSnapshotBeforeUpdate(prevProps, prevState) {
+        if (prevState.unreadChannelIds.length === this.props.unreadChannelIds.length &&
+            prevState.unreadChannelIds.includes(this.props.currentChannel.id) &&
+            this.props.orderedChannelIds[0].type === 'unreads'
+        ) {
+            const newOrderedChannelIds = this.props.orderedChannelIds;
+            newOrderedChannelIds[0].items = prevState.unreadChannelIds;
+
+            return {
+                unreadChannelIds: prevState.unreadChannelIds || [],
```

Comment:
why the empty array at the end? it should be an array already

Reply:
I initially thought to protect it since that prop is not required but since that value is from the selector that always returns an array, you're right about that.
Now removed.